x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/60_semantic_text_inference_update.yml

```
@@ -0,0 +1,612 @@
+          - >
+           {
+             "doc": {
+               "sparse_field": "sparse data 1",
+               "object_source.sparse_field": "sparse data 2",
+               "dense_field": "dense data 1",
+               "object_source.dense_field": "dense data 2"
+             }
+           }
+
+  - do:
+      get:
+        index: test-copy-to-index
+        id: doc_1
+
+  - match: { _source.sparse_field.text: "sparse data 1" }
+  - length: { _source.sparse_field.inference.chunks: 2 }
+  - match: { _source.sparse_field.inference.chunks.0.text: "sparse data 2" }
+  - match: { _source.sparse_field.inference.chunks.1.text: "sparse data 1" }
+  - match: { _source.dense_field.text: "dense data 1" }
+  - length: { _source.dense_field.inference.chunks: 2 }
+  - match: { _source.dense_field.inference.chunks.0.text: "dense data 1" }
+  - match: { _source.dense_field.inference.chunks.1.text: "dense data 2" }
+
+  - do:
+      update:
+        index: test-copy-to-index
+        id: doc_1
+        body:
+          doc:
+            {
+              "sparse_field": "sparse data 1",
+              "object_source.sparse_field": "sparse data 3",
+              "dense_field": "dense data 1",
+              "object_source.dense_field": "dense data 3"
+            }
+
+  - do:
+      get:
+        index: test-copy-to-index
+        id: doc_1
+
+  - match: { _source.sparse_field.text: "sparse data 1" }
+  - length: { _source.sparse_field.inference.chunks: 2 }
+  - match: { _source.sparse_field.inference.chunks.0.text: "sparse data 3" }
+  - match: { _source.sparse_field.inference.chunks.1.text: "sparse data 1" }
+  - match: { _source.dense_field.text: "dense data 1" }
+  - length: { _source.dense_field.inference.chunks: 2 }
+  - match: { _source.dense_field.inference.chunks.0.text: "dense data 1" }
+  - match: { _source.dense_field.inference.chunks.1.text: "dense data 3" }
+
+---
+"Updates fail when the updated value is invalid":
+  - do:
+      index:
+        index: test-index
+        id: doc_1
+        body:
+          sparse_field: "inference test"
+          dense_field: "another inference test"
+          non_inference_field: "non inference test"
+
+  - do:
+      get:
+        index: test-index
+        id: doc_1
+
+  - match: { _source.sparse_field.text: "inference test" }
+  - match: { _source.sparse_field.inference.chunks.0.text: "inference test" }
+  - match: { _source.dense_field.text: "another inference test" }
+  - match: { _source.dense_field.inference.chunks.0.text: "another inference test" }
+  - match: { _source.non_inference_field: "non inference test" }
+
+  - do:
+      bulk:
+        index: test-index
+        body:
+          - '{"update": {"_id": "doc_1"}}'
+          - '{"doc":{"sparse_field": [{"key": "value"}], "dense_field": [{"key": "value"}]}}'
+
+  - match: { errors: true }
+  - match: { items.0.update.status: 400 }
+  - match: { items.0.update.error.reason: "/Invalid\\sformat\\sfor\\sfield\\s\\[(dense|sparse)_field\\].+/" }
```

I think the following format is clearer:

```suggestion
  - match: { items.0.update.error.reason: /Invalid\ format\ for\ field\ \[(dense|sparse)_field\].+/ }
```

---

Done, although I had to keep the double backslashes and quotes to keep the YAML tests happy